File: backend/app/schemas/interview.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Literal
from typing import Optional

class EvaluationRequest(BaseModel):
    interview_type: Literal[
        "technical",
        "behavioral",
    ]
    topic: str | None = None
    difficulty: str
    question: str
    candidate_answer: str


class EvaluationResponse(BaseModel):
    question: str

    candidate_answer: str

    overall_score: float

    strengths: list[str]
    weaknesses: list[str]
    suggestions: list[str]

    semantic_score: float | None = None
    technical_score: float | None = None
    completeness_score: float | None = None
    communication_score: float | None = None

    professionalism_score: float | None = None
    leadership_score: float | None = None
    problem_solving_score: float | None = None
    star_score: float | None = None

    ideal_answer: str | None = None


class QuestionRequest(BaseModel):

    interview_type: Literal[
        "technical",
        "behavioral",
    ]
    topic: str | None = None
    difficulty: str


class QuestionResponse(BaseModel):
    interview_type: str
    topic: str | None = None
    difficulty: str
    question: str


class IdealAnswerRequest(BaseModel):
    question: str


class IdealAnswerResponse(BaseModel):
    ideal_answer: str

class InterviewHistoryItem(BaseModel):
    id: int
    interview_type: Literal[
        "technical",
        "behavioral",
    ]
    topic: str | None = None
    difficulty: str
    overall_score: float
    created_at: datetime

    model_config = ConfigDict(
        from_attributes=True,
    )

File: frontend/src/types/interview.ts
export interface Interview {
  id: number;

  user_id: number;

  topic: string;
  difficulty: string;
  interview_type: "technical" | "behavioral";

  question: string;
  ideal_answer: string;
  user_answer: string;

  semantic_score?: number;
  technical_score?: number;
  completeness_score?: number;
  communication_score?: number;
  overall_score: number;

  professionalism_score?: number;
  leadership_score?: number;
  problem_solving_score?: number;
  star_score?: number;

  feedback: string;
  status: string;
  created_at: string;
}

export interface QuestionRequest {
  interview_type: "technical" | "behavioral";

  topic?: string;

  difficulty: string;
}

export interface QuestionResponse {
  interview_type: "technical" | "behavioral";

  topic?: string;

  difficulty: string;

  question: string;
}

export interface EvaluationRequest {
  interview_type: "technical" | "behavioral";

  topic?: string;

  difficulty: string;

  question: string;

  candidate_answer: string;
}

export interface EvaluationResponse {
  question: string;

  candidate_answer: string;

  overall_score: number;

  strengths: string[];
  weaknesses: string[];
  suggestions: string[];

  // Technical interview fields
  semantic_score?: number;
  technical_score?: number;
  completeness_score?: number;
  communication_score?: number;

  // Behavioral interview fields
  professionalism_score?: number;
  leadership_score?: number;
  problem_solving_score?: number;
  star_score?: number;

  ideal_answer?: string;
}


export interface InterviewHistoryItem {
  id: number;
  interview_type: "technical" | "behavioral";
  topic?: string;
  difficulty: string;
  overall_score: number;
  created_at: string;
  status: string;
}

export interface ScoreBreakdownResult {

  overall_score: number;

  semantic_score?: number;
  technical_score?: number;
  completeness_score?: number;
  communication_score?: number;

  professionalism_score?: number;
  leadership_score?: number;
  problem_solving_score?: number;
  star_score?: number;
}